lib/spack/spack/compiler.py

```diff
@@ -406,6 +406,40 @@ def c11_flag(self):
                                       "the C11 standard",
                                       "c11_flag")
 
+    #
+    # Recompute the compiler version for a compiler
+    # Take advantage of modules to be cross-platform
+    # Not a class method
```

Comment:
I should expand on that. Most of the methods to get version info are class methods, but this one is not (because it needs access to the compiler object to setup the modules to be able to call it properly).